sql/cagg_api.sql

```diff
@@ -0,0 +1,179 @@
+-- This file and its contents are licensed under the Apache License 2.0.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-APACHE for a copy of the license.
+
+-- Helper function to get information about a hypertable and the
+-- primary dimension.
+CREATE OR REPLACE FUNCTION _timescaledb_functions.get_hypertable_info(
+       hypertable REGCLASS,
+       column_type REGTYPE
+) RETURNS integer AS
+$body$
+DECLARE
+   info RECORD;
+BEGIN
+   SELECT ht.id AS hypertable_id,
+          di.column_type::regtype,
+          EXISTS(SELECT FROM _timescaledb_catalog.continuous_agg where raw_hypertable_id = ht.id) AS has_cagg
+     INTO info
+     FROM _timescaledb_catalog.hypertable ht
+     JOIN _timescaledb_catalog.dimension di ON ht.id = di.hypertable_id
+    WHERE format('%I.%I', schema_name, table_name)::regclass = hypertable
+      AND di.interval_length IS NOT NULL;
+
+   IF info IS NULL THEN
+      RAISE 'table "%" is not a hypertable', hypertable
+      USING ERRCODE = 'object_not_in_prerequisite_state';
+   END IF;
+
+   IF NOT info.has_cagg THEN
+      RAISE 'hypertable "%" has no continuous aggregate', hypertable
+      USING HINT = 'Define a continuous aggregate for the hypertable to read invalidations.',
+            ERRCODE = 'object_not_in_prerequisite_state';
+   END IF;
```

Comment:
Do we need to care that this function can only be called for hypertables that have caggs? Why not just allow calling it for all hypertables?

Reply:
As mentioned above, this is function is only used internally and hence many checks that are shared between the overloaded versions are moved to this function.

Also added a comment to the tests to make it clear that this is not part of the official API.